ote_sdk/ote_sdk/entities/label_schema.py

```diff
@@ -540,9 +478,71 @@ def get_group_containing_label(self, label: LabelEntity) -> Optional[LabelGroup]
         return None
 
     def get_labels_exclusive_to(self, label: LabelEntity) -> List[LabelEntity]:
-        """Returns a list of labels that are exclusive to the passed label"""
-        self.__get_label(label)
-        return list(self.exclusivity_graph.neighbors(label))
+        """Returns a list of labels that are exclusive to the passed label."""
+        if label.is_empty:
+            exclusive_labels = self.__get_exclusivity_for_empty_label(label=label)
+        else:
+            exclusive_labels = self.__get_exclusivity_recursion(label=label)
+        return exclusive_labels
+
+    def __get_exclusivity_recursion(
+        self, label: LabelEntity, add_empty: bool = True
+    ) -> List[LabelEntity]:
+        """
+        Recursively computes all labels exclusive to a non-empty label. A label is exclusive with:
+        - All labels in the same group
+        - All children of labels in the same group
+        - All labels in the same group as any of the label's ancestors
+        - All children of labels in the same group as any of the label's ancestors
+        - All empty labels that are not descendants of the label
+
+        :param label: The label to get exclusive labels for
+        :param add_empty: If set to True, adds all empty labels that are not descendants of the label. This is only
+        needed for the first recursion iteration.
+        :return: List of labels exclusive to the label.
+        """
+        output = []
+
+        # Add all labels in the same group
+        siblings = self.get_siblings_in_group(label)
+        output += siblings
+
+        # Add all children of labels in the same group
+        for sibling in siblings:
+            output += self.get_children(sibling)
+
+        # Do the same for the parent of the label
+        parent = self.get_parent(label)
+        if parent is not None:
+            output += self.__get_exclusivity_recursion(parent, add_empty=False)
+
+        # Add all empty labels that are not descendants of the label
+        if add_empty:
+            descendants = self.get_descendants(label)
```

Comment:
Descendants and ancestors, no?

Reply:
This has the assumption built in that empty labels don't have descendants.

Reply:
Fair assumption, please add as a comment